File: solutions/typescript/space-age/1/space-age.ts
export type Planet = 'mercury' |
                     'venus'   |
                     'earth'   |
                     'mars'    |
                     'jupiter' |
                     'saturn'  |
                     'uranus'  |
                     'neptune';

export function age(planet: Planet, seconds: number): number {
  let age_on_earth = seconds / 31557600;

  switch(planet) {
    case 'mercury':
      return round2Digits(age_on_earth / 0.2408467);
    case 'venus':
      return round2Digits(age_on_earth / 0.61519726);
    case 'earth':
      return round2Digits(age_on_earth);
    case 'mars':
      return round2Digits(age_on_earth / 1.8808158);
    case 'jupiter':
      return round2Digits(age_on_earth / 11.862615);
    case 'saturn':
      return round2Digits(age_on_earth / 29.447498);
    case 'uranus':
      return round2Digits(age_on_earth / 84.016846);
    case 'neptune':
      return round2Digits(age_on_earth / 164.79132);
  }
}

function round2Digits(age: number): number {
  return Math.round(age * 100) / 100;
}

File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const age_on_earth = @as(f64, @floatFromInt(seconds)) / 31557600;
        
        return switch(self) {
            Planet.mercury => age_on_earth / 0.2408467,
            Planet.venus => age_on_earth / 0.61519726,
            Planet.earth => age_on_earth / 1.0,
            Planet.mars => age_on_earth / 1.8808158,
            Planet.jupiter => age_on_earth / 11.86261,
            Planet.saturn => age_on_earth / 29.447498,
            Planet.uranus => age_on_earth / 84.016846,
            Planet.neptune => age_on_earth / 164.79132,
        };
    }
};
